pynequa/core.py

```diff
@@ -8,17 +10,47 @@ class Sinequa(API):
         Sinequa API Client for Python
 
         Attributes:
+            access_token(str): authentication token for Sinequa
+            base_url(str): base URL for hosted Sinequa instance
             app_name(str): name of Sinequa app
             query_name(str): name of search query web service
     '''
-    app_name: str
-    query_name: str
 
-    def __init__(self, config: Dict) -> None:
-        super().__init__(config)
-        self.app_name = config["app_name"]  # name of application
+    def __init__(
+            self,
+            access_token: str,
+            base_url: str,
+            app_name: str = "vanilla-search",
+            query_name: str = "query",
+    ) -> None:
+        super().__init__(access_token=access_token, base_url=base_url)
```

Comment:
@anisbhsl Can we add nonetype check during construction as well? I found -- while using `larch` -- that without the validation fo None type, the downstream pipeline fails in later stages. Good to have in constructor itself for now.

Reply:
It is now being done in API class. It raises `ValueError` exception. Sorry for the trouble due to missing implementation.